### PR TITLE
ocamlPackages.ocaml_intrinsics: 0.15.0 → 0.15.2 and related fixes

### PR DESCRIPTION
**Pull request by vbgl**
###### Description of changes

- Disable `async_smtp` for OCaml < 4.12.
- Move `patdiff` inside `ocamlPackages` as it is also a library, built for a specific version of OCaml (this change fixes `bonsai` for non-default versions of OCaml).
- The update of `ocaml_intrinsics` make it available on ARM architectures.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bcc32** [comment]
Result of `nixpkgs-review pr 182694` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>comby</li>
    <li>flitter</li>
    <li>ligo</li>
  </ul>
</details>

**Review by bcc32 — APPROVED** [review]
LGTM, thanks.

File: pkgs/development/ocaml-modules/janestreet/0.15.nix [hunk 1]
```diff
@@ -132,6 +131,7 @@ with self;
   async_smtp = janePackage {
     pname = "async_smtp";
     hash = "1m00j7wcb0blipnc1m6by70gd96a1k621b4dgvgffp8as04a461r";
+    minimumOCamlVersion = "4.12";
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/janestreet/0.15.nix:134]: `minimalOCamlVersion` ?

**Comment by vbgl** [comment]
@GrahamcOfBorg build flitter ligo comby

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -4315,8 +4315,6 @@ with pkgs;
     inherit (darwin.apple_sdk.frameworks) Security;
   };
 
-  patdiff = callPackage ../tools/misc/patdiff { };
```
Comment (aij) [inline pkgs/top-level/all-packages.nix:4318]: Could we keep this as a top-level alias like we do for other CLI tools that happen to be written in OCaml?

Removing it completely is a breaking change for folks who are using it:

```
error: undefined variable 'patdiff'
```

Here's a PR: https://github.com/NixOS/nixpkgs/pull/185003/

**Pull request merged**